packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,623 @@
+		});
+	}
+
+	/**
+	 * Returns the start side of a direction - left for LTR, right for RTL
+	 *
+	 * @protected
+	 */
+	get directionStart() {
+		return this.effectiveDir === "rtl" ? "right" : "left";
+	}
+
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 *
+	 * @private
+	 */
+	_drawDefaultTickmarks(step, max, min) {
+		if (!this.tickmarks || !this.step) {
+			return;
+		}
+
+		// Convert number values to strings to let the CSS do calculations better
+		// rounding/subpixel behavior" and the most precise tickmarks distribution
+		const maxStr = String(max);
+		const minStr = String(min);
+		const stepStr = String(step);
+		const tickmarkWidth = "1px";
+
+		// There is a CSS bug with the 'currentcolor' value of a CSS gradient that does not
+		// respect the variable for more than one theme. It has to be set here for now.
+		const currentTheme = getTheme();
+		const currentColor = this.TICKMARK_COLOR_MAP[currentTheme];
+
+		this._tickmarksAmount = `${maxStr - minStr} / ${stepStr}`;
+		this._hiddenTickmarks = false;
+
+		// Transparent CSS gradient background
+		const tickmarksGradientBase = `linear-gradient(to right, ${currentColor} ${tickmarkWidth}, transparent 0) `;
+
+		// Draw the tickmarks as a patern over the gradient background
+		const tickmarksGradientdPattern = `0 center / calc((100% - ${tickmarkWidth}) / (${this._tickmarksAmount})) 100% repeat-x`;
+
+		// Combine to get the complete CSS background gradient property value
+		this._tickmarksBackground = `${tickmarksGradientBase + tickmarksGradientdPattern}`;
+
+		// If labelsInterval is specified draw labels for the necessary tickmarks
+		if (this.labelInterval > 0) {
+			this._drawDefaultLabels();
+		}
+	}
+
+	/**
+	 * Calculates the labels amout, width and text and creates them
+	 *
+	 * @private
+	 */
+	_drawDefaultLabels() {
+		const labelInterval = this.labelInterval;
+		const step = this.step;
+		const newNumberOfLabels = (this.max - this.min) / (step * labelInterval);
+
+		// If the required labels are already rendered
+		if (newNumberOfLabels === this._oldNumberOfLabels) {
+			return;
+		}
+
+		this._oldNumberOfLabels = newNumberOfLabels;
+		this._labelWidth = 100 / newNumberOfLabels;
+		this._labelItems = [];
+
+		// If the step value is not a round number get its precision
+		const stepPrecision = SliderBase._getDecimalPrecisionOfNumber(step);
+
+		// numberOfLabels below can be float so that the "distance betweenlabels labels"
+		// calculation to be precize (exactly the same as the distance between the tickmarks).
+		// That's ok as the loop stop condition is set to an integer, so it will practically
+		// "floor" the number of labels anyway.
+		for (let i = 0; i <= newNumberOfLabels; i++) {
+			// Format the label numbers with the same decimal precision as the value of the step property
+			const labelItemNumber = ((i * step * labelInterval) + this.min).toFixed(stepPrecision);
+			this._labelItems.push(document.createTextNode(labelItemNumber));
```

Comment:
Do you really need to create text nodes, would not just using the labelItemNumber sufficient?